providers/runtime.go

```diff
@@ -130,7 +130,7 @@ func (r *Runtime) UseProvider(id string) error {
 func (r *Runtime) AddConnectedProvider(c *ConnectedProvider) {
 	r.providers[c.Instance.ID] = c
 	r.schema.Add(c.Instance.Name, c.Instance.Schema)
-	r.schema.refresh()
+	r.schema.unsafeRefresh()
```

Comment:
why is it safe to call `unsafeRefresh` without holding the lock

Reply:
it's not, following the description and where this is located, it's an oversight